clientdb/sidecar.go

```diff
@@ -0,0 +1,139 @@
+package clientdb
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	"github.com/lightninglabs/pool/sidecar"
+	"go.etcd.io/bbolt"
+)
+
+var (
+	// ErrNoSidecar is the error returned if no sidecar with the given
+	// multisig pubkey exists in the store.
+	ErrNoSidecar = errors.New("no sidecar found")
+
+	// sidecarsBucketKey is a bucket that contains all sidecars that are
+	// currently pending or completed. This bucket is keyed by the multisig
```

Comment:
Right, if for whatever reason a provider decides to not use random IDs then it could lead to collisions. They need to be unique on the asker's/provider's side. But if a recipient gets tickets from multiple providers, it _could_ create a collision. Added the provider's signing pubkey to the key as well.